pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -125,6 +125,9 @@ self: super: {
   ### END HASKELL-LANGUAGE-SERVER SECTION ###
   ###########################################
 
+  # needed for ghcup
+  haskus-utils-variant = dontCheck super.haskus-utils-variant;
```

Comment:
Could you leave a comment about why this needs to be `dontCheck`?  Are the tests incorrect and need to be fixed upstream (in which case, can you report this upstream)?  Or are the tests trying to do something weird that doesn't work in the nix sandbox?

Reply:
I was going to do the same PR yesterday, and opened this issue: https://github.com/haskus/packages/issues/54

I suggest linking it in a comment above this line.

Reply:
Seems like there is a new issue in the new release (3.5) that supposedly fixes the issue:

```
Running 1 test suites...
Test suite tests: RUNNING...

==============================
TASTY
==============================
utils-variant
  Variant
    get by index (match):               OK
      +++ OK, passed 1 test.
    get by index (dont' match):         OK
      +++ OK, passed 1 test.
    pattern V: set:                     OK
      +++ OK, passed 1 test.
    pattern V: match:                   OK
      +++ OK, passed 1 test.
    pattern V: match2:                  OK
      +++ OK, passed 1 test.
    pattern V: type application:        OK
      +++ OK, passed 1 test.
    get by type (match):                OK
      +++ OK, passed 1 test.
    get by type (don't match):          OK
      +++ OK, passed 1 test.
    variant equality (match):           OK
      +++ OK, passed 1 test.
    variant equality (don't match):     OK
      +++ OK, passed 1 test.
    update by index (match):            OK
      +++ OK, passed 1 test.
    update by index (don't match):      OK
      +++ OK, passed 1 test.
    update by type (match):             OK
      +++ OK, passed 1 test.
    update by type (don't match):       OK
      +++ OK, passed 1 test.
    update/fold by index (match):       OK
      +++ OK, passed 1 test.
    update/fold by index (don't match): OK
      +++ OK, passed 1 test.
    Convert into tuple:                 OK
      +++ OK, passed 1 test.
    Convert single variant:             OK
      +++ OK, passed 1 test.
    Lift Either: Left:                  OK
      +++ OK, passed 1 test.
    Lift Either: Right:                 OK
      +++ OK, passed 1 test.
    To Either: Left:                    OK
      +++ OK, passed 1 test.
    To Either: Right:                   OK
      +++ OK, passed 1 test.
    popVariantHead (match):             OK
      +++ OK, passed 1 test.
    popVariantHead (don't match):       OK
      +++ OK, passed 1 test.
    popVariantAt (match):               OK
      +++ OK, passed 1 test.
    popVariantAt (don't match):         OK
      +++ OK, passed 1 test.
    popVariant (match):                 OK
      +++ OK, passed 1 test.
    popVariant (match):                 OK
      +++ OK, passed 1 test.
    popVariant (don't match):           OK
      +++ OK, passed 1 test.
    prependVariant:                     OK
      +++ OK, passed 1 test.
    appendVariant:                      OK
      +++ OK, passed 1 test.
    alterVariant:                       OK
      +++ OK, passed 1 test.
    alterVariant:                       OK
      +++ OK, passed 1 test.
    traverseVariant:                    OK
      +++ OK, passed 1 test.
    traverseVariant:                    OK
      +++ OK, passed 1 test.
    liftVariant:                        OK
      +++ OK, passed 1 test.
    splitVariant:                       OK
      +++ OK, passed 1 test.
    splitVariant2:                      OK
      +++ OK, passed 1 test.
    toCont:                             OK
      +++ OK, passed 1 test.
  EADT
    eadtPattern: match:                 OK
      +++ OK, passed 1 test.
    eadtInfixPattern: match:            OK
      +++ OK, passed 1 test.
    catamorphism: constraint:           OK
      +++ OK, passed 1 test.
    anamorphism: numbersTo:             OK
      +++ OK, passed 1 test.
    apomorphism: numbersToMin5:         OK
      +++ OK, passed 1 test.

All 44 tests passed (0.01s)

==============================
DOCTEST
==============================
src/lib/Haskus/Utils/EADT.hs:59: failure in expression `instance Eq a => Eq1 (ConsF a) where liftEq cmp (ConsF a e1) (ConsF b e2) = a == b && cmp e1 e2'
expected:
 but got:
          ^
          <interactive>:53:10: error: [GHC-39999]
              • Could not deduce ‘Eq (ConsF a a1)’
                  arising from the head of a quantified constraint
                  arising from the superclasses of an instance declaration
                from the context: Eq a
                  bound by the instance declaration at <interactive>:53:10-30
                or from: Eq a1
                  bound by a quantified context at <interactive>:53:10-30
              • In the instance declaration for ‘Eq1 (ConsF a)’

src/lib/Haskus/Utils/EGADT.hs:52: failure in expression `:t y'
expected: y :: EGADT '[LamF, AppF, VarF] Int
 but got: y :: EGADT [LamF, AppF, VarF] Int
                     ^

src/lib/Haskus/Utils/VariantF.hs:80: failure in expression `instance Eq a => Eq1 (ConsF a) where liftEq cmp (ConsF a e1) (ConsF b e2) = a == b && cmp e1 e2'
expected:
 but got:
          ^
          <interactive>:129:10: error: [GHC-39999]
              • Could not deduce ‘Eq (ConsF a a1)’
                  arising from the head of a quantified constraint
                  arising from the superclasses of an instance declaration
                from the context: Eq a
                  bound by the instance declaration at <interactive>:129:10-30
                or from: Eq a1
                  bound by a quantified context at <interactive>:129:10-30
              • In the instance declaration for ‘Eq1 (ConsF a)’

src/lib/Haskus/Utils/Variant.hs:334: failure in expression `:t y'
expected: y :: V '[Double, Int, String, Float]
 but got: y :: V [Double, Int, String, Float]
                 ^

src/lib/Haskus/Utils/Variant.hs:445: failure in expression `:t v'
expected: v :: V '[(), String, Float]
 but got: v :: V [(), String, Float]
                 ^

src/lib/Haskus/Utils/Variant.hs:677: failure in expression `fromVariant @Double x'
expected:
          ... error:
          ... Double not found in list:
          ... '[Int, String, Int]
          ...
 but got:
          <interactive>:735:1: error: [GHC-64725]
                                     ^
              • Double not found in list:
                 [Int, String, Int]
              • In the expression: fromVariant @Double x
                In an equation for ‘it’: it = fromVariant @Double x

          <interactive>:735:1: error: [GHC-64725]
              • Double not found in list:
                 [Int, String, Int]
              • In the expression: fromVariant @Double x
                In an equation for ‘it’: it = fromVariant @Double x

Examples: 228  Tried: 176  Errors: 0  Failures: 6

==============================
SUMMARY
==============================
Test suite tests: FAIL
Test suite logged to: dist/test/haskus-utils-variant-3.5-tests.log
0 of 1 test suites (0 of 1 test cases) passed.
```